src/sentry/static/sentry/app/api.tsx

```diff
@@ -58,7 +84,28 @@ export function paramsToQueryArgs(params) {
   return p;
 }
 
+// TODO: move this somewhere
+type APIRequestMethod = 'POST' | 'GET' | 'DELETE' | 'PUT';
+
+type FunctionCallback = (...args: any[]) => void;
+
+type RequestCallbacks = {
+  success?: FunctionCallback;
+  complete?: FunctionCallback;
+  error?: FunctionCallback;
+};
```

Comment:
I like to ideally refine these function callbacks. But I'm unable to find the right callback signatures.